src/runtime/policy/tests/policy-test.ts

```diff
@@ -808,9 +808,97 @@ policy MyPolicy {
          /* canWriteSuperset = */manifestPerson,
          /* canReadSubset = */personSubset);
-       assert.deepEqual(ingressValidation.getMaxReadType(subsetTypeVar), subsetTypeVar);
+       const expectedTypeVar = TypeVariable.make(
+         '',
+         /* canWriteSuperset = */personSubset,
+         /* canReadSubset = */personSubset);
+       assert.deepEqual(
+         ingressValidation.getMaxReadType(subsetTypeVar), expectedTypeVar);
      });
 
+  it('updates write w/ max read type that is consistent with read', async() => {
+    const manifest = await Manifest.parse(`
+      schema A
+        a: Text
+        b: Text
+        c: Text
+        d: Text
+
+      policy P {
+        from A access {a, b}
+      }
+    `);
+    const ingressValidation = new IngressValidation(manifest.policies);
+    const entityType = new EntityType(manifest.schemas['A']);
+    // Create a type variable that captures writes `A {a, d}`
+    const writeA = (await Manifest.parse(`
+      schema A
+        a: Text
+        d: Text
+    `)).schemas['A']
+    const writeEntityType = new EntityType(writeA);
+    const typeVar = TypeVariable.make(
+         '',
+         /* canWriteSuperset = */null,
+         /* canReadSubset = */writeEntityType);
+    // Note that the max read type for schema A is `A {a, b}`. However, the
+    // expected max read type variable should have `A {a}` for writeSuperset
+    // so that it is consistent with the write `A {a, d}`.
+    const maxReadA = (await Manifest.parse(`
+      schema A
+        a: Text
+    `)).schemas['A'];
+    deleteFieldRecursively(maxReadA, 'location', {replaceWithNulls: true});
+    const maxReadEntityType = new EntityType(maxReadA);
+    const expected = TypeVariable.make(
+      '',
+      /* canWriteSuperset = */maxReadEntityType,
+      /* canReadSubset = */writeEntityType);
+    assert.deepEqual(
+      ingressValidation.getMaxReadType(typeVar), expected);
+  });
+
+  it('updates write w/ max read type that is consistent with read (inline)', async() => {
+    const manifest = await Manifest.parse(`
+      schema A
+        foo: inline Foo {a: Text, b: Text, c: Text, d: Text}
+
+      policy P {
+        from A access {
+          foo {a, b}
+        }
+      }
+    `);
+    const ingressValidation = new IngressValidation(manifest.policies);
+    const entityType = new EntityType(manifest.schemas['A']);
+    // Create a type variable that captures writes `A { foo {a, d} }
+    const writeA = (await Manifest.parse(`
+      schema A
+        foo: inline Foo {a: Text, d: Text}
+    `)).schemas['A']
+    const writeEntityType = new EntityType(writeA);
+    const typeVar = TypeVariable.make(
+         '',
+         /* canWriteSuperset = */null,
+         /* canReadSubset = */writeEntityType);
+    // The expected max read type variable should have `A { foo {a} }` for writeSuperset.
+    const maxReadA = (await Manifest.parse(`
+      schema A
+        // Should be the following when b/175169555 is fixed:
```

Comment:
maybe rephrase to `// TODO(b/...): ...` so it's easier to find when the bug is fixed?

Reply:
Done.